eclair-core/src/test/scala/fr/acinq/eclair/router/ChannelRangeQueriesSpec.scala

```diff
@@ -127,4 +129,163 @@ class ChannelRangeQueriesSpec extends FunSuite {
     assert(Router.computeFlag(channels)(ef.shortChannelId, None, None, false) === (INCLUDE_CHANNEL_ANNOUNCEMENT | INCLUDE_CHANNEL_UPDATE_1 | INCLUDE_CHANNEL_UPDATE_2))
     assert(Router.computeFlag(channels)(ef.shortChannelId, None, None, true) === (INCLUDE_CHANNEL_ANNOUNCEMENT | INCLUDE_CHANNEL_UPDATE_1 | INCLUDE_CHANNEL_UPDATE_2 | INCLUDE_NODE_ANNOUNCEMENT_1 | INCLUDE_NODE_ANNOUNCEMENT_2))
   }
+
+  def makeShortChannelIds(height: Int, count: Int): List[ShortChannelId] = (0 until count).map(c => ShortChannelId(height, c, 0)).toList
+
+  def validate(chunk: ShortChannelIdsChunk) = {
+    require(!chunk.shortChannelIds.exists(!Router.keep(chunk.firstBlock, chunk.numBlocks, _)))
+  }
+
+  // check that chunks do not overlap and contain exactly the ids they were built from
+  def validate(ids: SortedSet[ShortChannelId], firstBlockNum: Long, numberOfBlocks: Long, chunks: List[ShortChannelIdsChunk]): Unit = {
+
+    @tailrec
+    def noOverloap(chunks: List[ShortChannelIdsChunk]): Boolean = chunks match {
+      case Nil => true
+      case a :: b :: _ if b.firstBlock < a.firstBlock + a.numBlocks => false
+      case _ => noOverloap(chunks.tail)
+    }
+
+    // aggregate ids from all chunks, to check that they match our input ids exactly
+    val chunkIds = SortedSet.empty[ShortChannelId] ++ chunks.map(_.shortChannelIds).flatten.toSet
+    val expected = ids.filter(Router.keep(firstBlockNum, numberOfBlocks, _))
+
+    if (expected.isEmpty) require(chunks == List(ShortChannelIdsChunk(firstBlockNum, numberOfBlocks, Nil)))
```

Comment:
This looks redundant with the tests below

Reply:
I still think this statement is unnecessary: the following lines re-test that (even for the case where `expected.isEmpty`)

Reply:
Yes but I'd like to keep it as intent here is really obvious